nms/app/views/domain-proxy/LogsList.js

```diff
@@ -0,0 +1,494 @@
+import Text from '../../theme/design-system/Text';
+import {REFRESH_INTERVAL} from '../../components/context/RefreshContext';
+import {colors} from '../../theme/default';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    margin: theme.spacing(3),
+    flexGrow: 1,
+  },
+  logsDirectionFilter: {
+    width: theme.spacing(19),
+  },
+  filterText: {
+    color: colors.primary.comet,
+  },
+  filterActionsWrapper: {
+    paddingRight: theme.spacing(4),
+  },
+  filterActions: {
+    textAlign: 'right',
+  },
+}));
+
+type LogsDirectionNullable = 'SAS' | 'DP' | 'CBSD' | null;
+
+type LogsDirectionFilterProps = {
+  value: LogsDirectionNullable,
+  onChange: (value: LogsDirectionNullable) => void,
+  selectProps: SelectProps,
+};
+
+function LogsDirectionFilter({
+  value,
+  onChange,
+  selectProps,
+}: LogsDirectionFilterProps) {
+  const classes = useStyles();
+
+  let parsedValue;
+  switch (value) {
+    case 'SAS':
+      parsedValue = 1;
+      break;
+    case 'DP':
+      parsedValue = 2;
+      break;
+    case 'CBSD':
+      parsedValue = 3;
+      break;
+    default:
+      parsedValue = 0;
+      break;
+  }
+
+  return (
+    <FormControl>
+      <Select
+        className={classes.logsDirectionFilter}
+        value={parsedValue}
+        onChange={({target}) => {
+          let newValue;
+          switch (parseInt(target.value)) {
+            case 1:
+              newValue = 'SAS';
+              break;
+            case 2:
+              newValue = 'DP';
+              break;
+            case 3:
+              newValue = 'CBSD';
+              break;
+            default:
+              newValue = null;
+              break;
+          }
+          onChange(newValue);
+        }}
+        input={<OutlinedInput />}
+        {...selectProps}>
+        <MenuItem value={0}>Any</MenuItem>
+        <MenuItem value={1}>SAS</MenuItem>
+        <MenuItem value={2}>DP</MenuItem>
+        <MenuItem value={3}>CBSD</MenuItem>
+      </Select>
+    </FormControl>
+  );
+}
+
+function LogsList() {
+  const classes = useStyles();
+
+  const params = useParams();
+  const networkId: string = nullthrows(params.networkId);
+
+  const tableRef = useRef(null);
+
+  const [isAutoRefreshing, setIsAutoRefreshing] = useState(true);
+  const {startDate, endDate, setStartDate, setEndDate} = useRefreshingDateRange(
+    isAutoRefreshing,
+    REFRESH_INTERVAL,
+    () => {
+      tableRef.current && tableRef.current.onQueryChange();
+    },
+  );
+  const [serialNumber, setSerialNumber] = useState<string>('');
+  const [fccId, setFccId] = useState<string>('');
+  const [from, setFrom] = useState<LogsDirectionNullable>(null);
+  const [to, setTo] = useState<LogsDirectionNullable>(null);
+  const [responseCode, setResponseCode] = useState<string | null>(null);
+  const [logName, setLogName] = useState<string>('');
+
+  const getDataFn = useCallback(
+    query => {
+      return new Promise(async (resolve, reject) => {
```

Comment:
We should not mix explicit promise creation and async functions.
We can just omit the  promise creation here and throw an error instead of calling reject.

Reply:
Well noticed! I've simplified it to use `async/await` only